### PR TITLE
Filter JS scope up to and including the inclusion of the Dart SDK

### PR DESCRIPTION
**Pull request by grouma**
The number of scopes to skip is not deterministic, so skipping the first two scopes leads to the inclusion of a number of non-useful properties in some scenarios. Instead, filter scopes until the inclusion of the Dart SDK. Nested scopes should contain application logic.

Related: https://github.com/dart-lang/sdk/issues/40774

### CODE REVIEW

File: dwds/lib/src/debugging/dart_scope.dart [hunk 1]
```diff
@@ -46,7 +43,28 @@ Future<List<Property>> visibleProperties(
   return allProperties;
 }
 
-/// Find the `this` in scope if it wasn't in the provided data from Chrome.
```
Comment (natebosch) [inline dwds/lib/src/debugging/dart_scope.dart]: Why was this line removed?

Comment (grouma) [reply dwds/lib/src/debugging/dart_scope.dart]: Unintentional. I'll add it back.
